### PR TITLE
Support redis as an alternative to memcached

### PR DESCRIPTION
**Pull request by JackDanger**
This has been in production for over a year and is working well. cc: @sinneduy

### CODE REVIEW

File: lib/gemstash/cli/setup.rb [hunk 1]
```diff
@@ -78,6 +78,7 @@ def ask_cache
         say_current_config(:cache_type, "Current cache")
         @config[:cache_type] = ask_with_default("Cache with what?", %w[memory memcached], "memory")
```
Comment (olleolleolle) [inline lib/gemstash/cli/setup.rb]:
```suggestion
        @config[:cache_type] = ask_with_default("Cache with what?", %w[memory memcached redis], "memory")
```

File: lib/gemstash/cli/setup.rb [hunk 2]
```diff
@@ -87,6 +88,13 @@ def ask_memcached_details
         @config[:memcached_servers] = servers
       end
 
+      def ask_redis_details
+        say_current_config(:redis_servers, "Current Redis servers")
+        servers = @cli.ask "What is the comma separated Redis servers? [localhost:6379]"
```
Comment (olleolleolle) [inline lib/gemstash/cli/setup.rb]:
```suggestion
        servers = @cli.ask "What is the comma-separated list of Redis servers? [localhost:6379]"
```

Comment (olleolleolle) [reply lib/gemstash/cli/setup.rb]: Well, I'd have to improve the wording of the other question, too.

**Comment by phanle** [comment]
Any update on this PR? Is there anything we could do (from Gusto) to make this change accepted?

**Comment by bronzdoc** [comment]
@phanle I think just apply the suggested changes by @olleolleolle

**Comment by phanle** [comment]
@bronzdoc updated and fixed tests. Let me know if you want to see more updates

**Comment by micheelengronne** [comment]
Is it possible to merge this MR ?

**Comment by indirect** [comment]
I'm not 100% sure about adding a hard dependency on the redis gem for an optional feature. I'm going to remove the hard dependency and merge, and people who need redis will need a Gemfile with both gemstash and redis in it. Open to arguments I should add it back, though. :)

File: gemstash.gemspec [hunk 3]
```diff
@@ -37,6 +37,7 @@ you push your own private gems as well."
   spec.add_runtime_dependency "faraday_middleware", "~> 0.10"
   spec.add_runtime_dependency "lru_redux", "~> 1.1"
   spec.add_runtime_dependency "puma", "~> 4.0"
+  spec.add_runtime_dependency "redis", "~> 3.3"
```
Comment (indirect) [inline gemstash.gemspec:40]:
```suggestion

```

**Pull request merged**